v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/schema/SourceSchemaReference.java

```diff
@@ -15,21 +15,45 @@
  */
 package com.google.cloud.teleport.v2.source.reader.io.schema;
 
-import com.google.auto.value.AutoValue;
+import com.google.auto.value.AutoOneOf;
+import com.google.cloud.teleport.v2.source.reader.io.schema.sources.CassandraSchemaReference;
+import com.google.cloud.teleport.v2.source.reader.io.schema.sources.JdbcSchemaReference;
 import java.io.Serializable;
-import javax.annotation.Nullable;
 
-/** Value class to enclose the database name and PG namespace. */
-@AutoValue
+/**
+ * Different sources employ varying styles for referencing a schema. For instance, a Postgres source
+ * uses a database name and namespace, while a Cassandra source uses a cluster name and keyspace. In
+ * the below Value class we enclose the schema reference. Note: In most of the cases an interface
+ * would be a better choice to a `OneOf` pattern. In this particular case though, it's not possible
+ * to have a common minimal interface that encodes getters and setters for properties like nameSpace
+ * or ClusterName. If we go that route, we would either have to compriamise on readability (like
+ * level1, level2 reference etc), or have a bloated interface. Hence, a oneOf of independent (but
+ * conceptually related) classes suits a better pattern to encode this information.
+ *
+ * @see: <a
+ *     heref=https://github.com/google/auto/blob/main/value/userguide/howto.md#-make-a-class-where-only-one-of-its-properties-is-ever-set>autoOneOf</a>
+ */
+@AutoOneOf(SourceSchemaReference.Kind.class)
 public abstract class SourceSchemaReference implements Serializable {
 
-  public abstract String dbName();
+  public enum Kind {
+    JDBC,
```

Comment:
this class will also need to be edited for every source.

Reply:
Similar to `DataSource`, the `SchemaReference` is a one-of for the actual source specific schema references.
Though we need to add a new Kind here, it's mostly just a few lines of change as the actual Cassandra, JDBC or other schemaReferences are entirely independent.

Reply:
A small note from our discussion today morning.

This tradeoff does weigh on compile time type safety and allows for internal plugability as against allowing external plugability - like say loading a jar for a source interface.

We can enhance this if we need external plugability. For example, when we get to allowing externally plugable sources, we could have a `Kind` that says external and have a more flexibale interface for externally plugabble sources. 
As we don't have the plugability requirements as yet, adding a bit of code for type safety would enhance reliability.

Reply:
Resolving as discussed.